data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,441 @@
+    :param pid: person_id
+    :param date_row:
+    :return:
+    """
+    check_pid_df = bq.query(
+        CHECK_PID_EXIST_QUERY.format(project=date_row.project_id,
+                                     dataset=date_row.dataset_id,
+                                     table=date_row.table,
+                                     pid=pid))
+    return bool(check_pid_df.get_value(0, 'count') > 0)
+
+
+def create_queries(project_id, ticket_number, pids_project_id, pids_dataset_id,
+                   pids_table):
+    """
+    Creates sandbox and truncate queries to run for EHR deactivated retraction
+
+    :param project_id: bq name of project
+    :param ticket_number: Jira ticket number to identify and title sandbox table
+    :param pids_project_id: deactivated ehr pids table in bq's project_id
+    :param pids_dataset_id: deactivated ehr pids table in bq's dataset_id
+    :param pids_table: deactivated pids table in bq's table name
+    :return: list of queries to run
+    """
+    queries_list = []
+    # Hit bq and receive df of deactivated ehr pids and deactivated date
+    deactivated_ehr_pids_df = bq.query(
+        DEACTIVATED_PIDS_QUERY.format(project=pids_project_id,
+                                      dataset=pids_dataset_id,
+                                      table=pids_table), project_id)
+
+    date_columns_df = get_date_info_for_pids_tables(project_id)
+    logging.info(
+        'Dataframe creation complete to loop through to create retraction queries'
+    )
+
+    # Loop through the deactivated PIDS df to create queries based on the retractions needed per PID table
+    logging.info(
+        'looping through deactivated pids df to create queries based on teh retractions needed per pid table'
+    )
+    for ehr_row in deactivated_ehr_pids_df.itertuples(index=False):
+        for date_row in date_columns_df.itertuples(index=False):
+            # Determine if dataset is deid to correctly pull pid or research_id and check if ID exists in dataset or if
+            # already retracted
+            if is_deid_dataset(date_row.dataset_id) is True:
+                pid = get_research_id(date_row.project_id, date_row.dataset_id,
+                                      ehr_row.person_id)
+            else:
+                pid = ehr_row.person_id
+
+            # Check if PID is in table
+            if check_pid_exist(pid, date_row) is True:
+                # Get or create sandbox dataset
+                sandbox_dataset = get_sandbox_dataset_id(date_row.dataset_id)
+                datasets_obj = bq.list_datasets(project_id)
+                datasets = [d.dataset_id for d in datasets_obj]
+                if sandbox_dataset not in datasets:
+                    logging.info('%s dataset does not exist, creating now' %
+                                 sandbox_dataset)
+                    create_sandbox_dataset(date_row.project_id,
+                                           date_row.dataset_id)
+
```

Comment:
I know you and Nishanth were wanting to have a `get_sandbox_dataset` function that guaranteed the sandbox dataset was created.  I was looking at the existing `create_sandbox_dataset` function and realized the reason you were having issues with using this function is because it has a bug.  Lines 20-24 should actually be:
```
def create_sandbox_dataset(project_id, dataset_id, raise_error=True):
    # Lines 16 - 19
    try:
        create_dataset(project_id=project_id,
                       dataset_id=sandbox_dataset_id,
                       friendly_name=friendly_name,
                       description=description,
                       overwrite_existing=False)
    except HttpError as err:
        if err.code != 409:
            raise
        elif raise_error:
            raise

    return sandbox_dataset_id
```
This allows `create_sandbox_dataset` to create a dataset if it doesn't exist (which is what it was trying and failing to do).  It also allows you to specify if you are just trying to ensure the dataset exists or if you want to raise the error.  The default is set to raise the error if the dataset already exists because that is how it was originally written.

You could make this change and replace lines 271-279 with:
```
    create_sandbox_dataset(date_row.project_id, date_row.dataset_id, raise_error=False)
```

Reply:
For now, since there is a ticket to fix `sandbox.py`, I have updated this to the new function that was merged last week `check_and_create_sandbox_dataset`.